processes/record_frbrizer.py

```diff
@@ -0,0 +1,120 @@
+from collections import defaultdict
+from lxml import etree
+import re
+
+from logger import create_log
+from mappings.oclc_bib import OCLCBibMapping
+from mappings.oclcCatalog import CatalogMapping
+from managers import DBManager, OCLCCatalogManager, RedisManager
+from model import Record
+from .record_buffer import RecordBuffer
+
+
+logger = create_log(__name__)
+
+
+class RecordFRBRizer:
+
+    def __init__(self, db_manager: DBManager):
+        self.db_manager = db_manager
+
+        self.oclc_catalog_manager = OCLCCatalogManager()
+        
+        self.redis_manager = RedisManager()
+        self.redis_manager.createRedisClient()
+
+        self.record_buffer = RecordBuffer(db_manager=self.db_manager)
+
+    def frbrize_record(self, record: Record) -> Record:
+        owi_to_oclc_numbers = self._add_works_for_record(record=record)
+        self._add_edition_records(owi_to_oclc_numbers=owi_to_oclc_numbers)
+
+        self.record_buffer.flush()
+
+        record.frbr_status = 'complete'
+
+        self.db_manager.session.add(record)
+        self.db_manager.session.commit()
+
+        logger.info(f'FRBRized record: {record}')
+
+        return record
+
+    def _add_works_for_record(self, record: Record) -> defaultdict:
```

Comment:
just a note - returning a defaultdict *can* result in surprising behavior downstream (eg, a key appearing to be in there when it's not actually). Doesn't seem like it'll be a problem with how it's currently used, but might be worth future-proofing by returning a plain dict

Reply:
interesting - good callout. I'll explore using a set of tuples instead. I think that it may be easier on memory and quicker